src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionTestCaseEventSender.cs

```diff
@@ -12,24 +13,61 @@ namespace Microsoft.VisualStudio.TestPlatform.CommunicationUtilities
 
     internal class DataCollectionTestCaseEventSender : IDataCollectionTestCaseEventSender
     {
+        private static readonly object obj = new object();
+
         private readonly ICommunicationManager communicationManager;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="DataCollectionTestCaseEventSender"/> class. 
         /// </summary>
-        public DataCollectionTestCaseEventSender() : this(new SocketCommunicationManager())
+        protected DataCollectionTestCaseEventSender() : this(new SocketCommunicationManager())
         {
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="DataCollectionTestCaseEventSender"/> class. 
         /// </summary>
         /// <param name="communicationManager">Communication manager.</param>
-        public DataCollectionTestCaseEventSender(ICommunicationManager communicationManager)
+        protected DataCollectionTestCaseEventSender(ICommunicationManager communicationManager)
         {
             this.communicationManager = communicationManager;
         }
 
+        /// <summary>
+        /// Gets the singleton instance of DataCollectionTestCaseEventSender.
+        /// </summary>
+        internal static DataCollectionTestCaseEventSender Instance { get; private set; }
```

Comment:
DataCollectionTestCaseEventSender is used to send events to DataCollector Process from TestExecution Process. We get DataCollector Port as arguments which is where we are instantiating object of this class and establish the socket communication. Later on, this class is used to send the events, which starts when the TestExecution is in progress. Therefore, this class is exposed as singleton.